test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -146,6 +147,37 @@ call_log_manager::process_call_log_entry(json call_log_entry)
             }
             break;
         }
+        case api_method::query_timestamp: {
+            /*
+             * Not having a valid connection is a fatal error since no other operations can happen
+             * without a connection.
+             */
+            if (_conn == nullptr)
```

Comment:
We keep checking for this, there must be a better way instead of duplicating the code.

Reply:
I can't think of a way to check it once. Happy to change it if there is a better way.

Reply:
@Sean04 do you think we could check if the `_conn` is `nullptr` once for all? Or at only one place?

Reply:
We want to do that check for all cases other than `wiredtiger_open`. Maybe we could do something like this `        if (_api_map.at(method_name) != api_method::wiredtiger_open && _conn == nullptr)` before the switch statement?

Reply:
Yea, sounds good Sean, thanks! I will make those changes as part of this PR.

Reply:
@etienneptl suggested removing the wiredtiger_open enum and initialising the connection in the call_log_manager constructor.
Let me know what you think @Sean04 
updated in [49dab20](https://github.com/wiredtiger/wiredtiger/pull/8145/commits/49dab2039d7eb8f662d98610c3e1886506aad07e)

Reply:
Okay so we won't bother with any wiredtiger_open call log entries?

Reply:
Yup.